tests/x11/sshxterm.pm

```diff
@@ -16,19 +16,22 @@ sub run() {
     mouse_hide(1);
     x11_start_program("xterm");
     script_run("ssh -XC root\@localhost xterm");
-    type_string "yes\n";
-    wait_idle 6;
+    my $ret = check_screen([qw/ssh-xterm-host-key-authentication ssh-password-prompt/]);
+    die "ssh did not respond in time or needle(s) not found" unless $ret;
+    # if ssh asks for authentication of the key accept it
+    if ($ret->{needle}->has_tag('ssh-xterm-host-key-authentication')) {
```

Comment:
There is a handy helper `testapi::match_has_tag`. You then don't need to store matched needle, just call `match_has_tag('ssh-xterm-host-key-authentication');`